sim/druid/talents.go

```diff
@@ -191,6 +194,106 @@ func (druid *Druid) applyGenesis() {
 	}
 }
 
+func (druid *Druid) applyEuphoria() {
+	if druid.Talents.Euphoria == 0 {
+		return
+	}
+
+	euphoriaSpellId := []int32{0, 81061, 81062}[druid.Talents.Euphoria]
+	euphoriaProcChancePct := []float64{0, 0.12, 0.24}[druid.Talents.Euphoria]
+	euphoriaManaGainPct := []float64{0, 0.08, 0.16}[druid.Talents.Euphoria]
+
+	// Mana return
+	manaMetrics := druid.NewManaMetrics(core.ActionID{
+		SpellID: euphoriaSpellId,
+	})
+
+	if druid.HasEclipseBar() {
+		druid.AddEclipseCallback(func(_ Eclipse, gained bool, sim *core.Simulation) {
+			if gained {
+				druid.AddMana(sim, druid.MaxMana()*euphoriaManaGainPct, manaMetrics)
+			}
+		})
+	}
+
+	// Double eclipse energy
+	euphoriaDummyAura := druid.GetOrRegisterAura(core.Aura{
```

Comment:
This aura doesnt do anything as far as I can see but is activated before the Wrath and Starfire apply effects are called and is deactivated on a cast complete which will come from the same spell that is activating the aura from the proc trigger. Dont know if that is what you intended or this is just wip stuff